### DISCUSSION TITLE
storyStoreV7 and Svelte

### DISCUSSION THREAD

**Discussion by dimfeld** [storybookjs/storybook#18216] [Help] [answered]
It appears that Storybook doesn't find any stories when using `storyStoreV7` and Svelte stories in CSF format, since the `StoryIndexGenerator` [skips Svelte files](https://github.com/storybookjs/storybook/blob/612b03d75cea8c1d82132243a2473765e8222fad/lib/core-server/src/utils/StoryIndexGenerator.ts#L51). I tried bypassing this and it broke immediately when trying to extract the stories, since it was using the default CSF code instead of the Svelte-aware CSF parser from `@storybook/addon-svelte-csf`. 

Will `addon-svelte-csf` eventually be moved into the core so that things will just work? Does `StoryIndexGenerator` need to have a pluggable CSF parser? Maybe something else?

This isn't blocking me from doing anything right now. Just curious since I'd like to take advantage of the new speed improvements coming in 6.5.

**Comment by IanVS** [comment, marked as answer]
This has been addressed in 7.0, which allows indexing svelte files.  The latest 3.0.0 prerelease versions of `@storybook/addon-svelte-csf` use the new story indexer added in 7.0.

**Comment by shilman** [comment]
Closing this as fxed!
